packages/celotool/src/lib/prometheus.ts

```diff
@@ -99,35 +103,51 @@ async function helmParameters(clusterConfig?: BaseClusterConfig) {
     `--set sidecar.imageTag=${sidecarImageTag}`,
     `--set prometheus.imageTag=${prometheusImageTag}`,
     `--set stackdriver_metrics_prefix=${prometheusImageTag}`,
+    `--set serviceAccount.name=${kubeServiceAccountName}`,
     // Stackdriver allows a maximum of 10 custom labels. kube-state-metrics
     // has some metrics of the form "kube_.+_labels" that provides the labels
     // of k8s resources as metric labels. If some k8s resources have too many labels,
     // this results in a bunch of errors when the sidecar tries to send metrics to Stackdriver.
     `--set-string includeFilter='\\{job=~".+"\\,${exclusions.join('\\,')}\\}'`,
   ]
-  if (clusterConfig) {
+  if ((clusterConfig) && (clusterConfig.cloudProvider) !== CloudProvider.GCP ) {
+    const clusterName = clusterConfig.clusterName
+    const cloudProvider = getCloudProviderPrefix(clusterConfig)
     params.push(
       `--set cluster=${clusterConfig.clusterName}`,
       `--set stackdriver_metrics_prefix=external.googleapis.com/prometheus/${clusterConfig.clusterName}`,
       `--set gcloudServiceAccountKeyBase64=${await getPrometheusGcloudServiceAccountKeyBase64(
-        clusterConfig
+        clusterName, cloudProvider
       )}`
     )
   } else {
+    // GCP
     const clusterName = fetchEnv(envVar.KUBERNETES_CLUSTER_NAME)
+    const gcloudProjectName = fetchEnv(envVar.TESTNET_PROJECT_NAME)
+    const cloudProvider = 'gcp'
+    const serviceAccountName = getServiceAccountName(clusterName, cloudProvider)
+    await createPrometheusGcloudServiceAccount(serviceAccountName, gcloudProjectName)
+    console.info(serviceAccountName)
+    const serviceAccountEmail = await getServiceAccountEmail(serviceAccountName)
     params.push(
       `--set cluster=${clusterName}`,
-      `--set stackdriver_metrics_prefix=external.googleapis.com/prometheus/${clusterName}`
+      `--set storageClassName=ssd`,
+      `--set stackdriver_metrics_prefix=external.googleapis.com/prometheus/${clusterName}`,
+      `--set serviceAccount.annotations.'iam\\\.gke\\\.io/gcp-service-account'=${serviceAccountEmail}`
```

Comment:
how do these changes affect clusters that don't have workload identity enabled? (I assume we have some not enabled?)

Reply:
So if for that cluster we set the `PROMETHEUS_GCE_SCRAPE_REGIONS` to some region, prometheus would fail to scrape every `scrape_time` and will log some trace warning about this, but it would work for the rest of scrape jobs.
If we do not setup any value for `PROMETHEUS_GCE_SCRAPE_REGIONS` in those cluster I think it should work as it has worked now. Anyway most of the existing gcp clusters have already enabled workload identity (manual process).